pkgs/tools/networking/curl-impersonate/update.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env nix-shell
```

Comment:
Are we doing anything complicated in here that would be not that easy with bash, curl and jq?

Reply:
I think you could write this in bash, but when I started trying to parse the makefile in bash the code got large and annoying real fast (so I switched to Python). If it's really desired, I can probably rewrite it in bash when I have time

Reply:
Now that we don't need to fetch other libraries the default nix-update $name might already be enough.

Reply:
We do need to fetch at least BoringSSL and NSS, as @lwthiker mentioned it being sensitive to those versions, and we would still have to parse the makefile to know where to name the sources for nghttp2 and brotli so that the build script can configure them

I'm not sure trying to avoid vendoring those two dependency source URLs is worth it either given it seems to require building them from source anyway, and if we use all of the vendored deps rather than some, automation is somewhat easier for package updates and maintenance burden is lower

I've rewritten the updater script in bash now, though, if that helps

Any thoughts @SuperSandro2000? (thanks you again for the reviews!)

Reply:
We want to use as little vendored libraries as possible because over time they are usually neglected and they never have nix specific patches which might be very important.

Reply:
I get that, but to avoid vendoring we would need to also seprately build compatible libs (with some of the curl-impersonate patches for at least boringssl) and bypass the curl-impersonate makefile entirely to write our own build logic (since it expects to build libs with custom options itself)

I'll assess how much work it would be sometime in the next few days to get an idea of what it would take to implement (including upfront difficulty in rewriting all of the relevant source files and in continued maintenance costs)

Also yeah as it is now the static build of NSS in this package is missing the patches to set an `NSS_LIB_DIR` (even if that doesn't really provide any value to curl-impersonate). I don't remember other Nix-specific patches this would be missing off the top of my head, but there may be one in boringssl to set a `NIX_SSL_DIR`, idk